### ISSUE TITLE
Update heading levels on the Repository page to eliminate missed levels

### ISSUE THREAD

**Issue by jvine** [projectblacklight/arclight#835]
Part of #820

Page example:
https://arclight-demo.projectblacklight.org/repositories/duke-rubenstein

- [x] Change `<h3><%= t('arclight.views.show.our_collections') %></h3>`  to `<h2>`
- [x] Change `<h5><%= link_to document.normalized_title, solr_document_path(document.id) %></h5>` to `<h3>`

Combined with #834, the final heading structure for the page will be:
```
h1. Repository name
  h2. Our Collections
     h3. Collection title 1
     h3. Collection title 2
```
